recipe/patches/0001_Fix_xmlErrorPtr.patch

```diff
@@ -0,0 +1,58 @@
+From a13d70254ef1c0788f584aeb5ec25392fe9f8497 Mon Sep 17 00:00:00 2001
+From: Steven <scrass@anaconda.com>
+Date: Thu, 8 Aug 2024 10:22:57 -0600
+Subject: [PATCH] Fix convention error with xmlErrorPtr caused by libxml2 API change
```

Comment:
if the libxml API changed, is there a possibility to updating to a pdal that's compatible with the new API?

Reply:
Possibly, but someone from PSUBS would most likely need to handle that.

Reply:
@danpetry 

```
Starting with `libxml2 2.12.0`, the API has changed the const-ness of the
`xmlError` pointers, which results in a build error due to a mismatched
type on `const xmlError *` and `xmlErrorPtr`.
This commit papers over the difference by using preprocessor conditionals.

Related libxml2 commit and issue:
- https://gitlab.gnome.org/GNOME/libxml2/-/commit/45470611b047db78106dcb2fdbd4164163c15ab7
- https://gitlab.gnome.org/GNOME/libxml2/-/commit/61034116d0a3c8b295c6137956adc3ae55720711
```

Reply:
oh, it's a snowflake package. ok fair

Reply:
It was not the best way to handle pinnings a year ago.
A plain `libxml2` in `run` is fine

Reply:
> It was not the best way to handle pinnings a year ago.

yeah this method was pushed by kai IIRC